frontend/class-opengraph-oembed.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Frontend
+ */
+
+/**
+ * Class WPSEO_OpenGraph_OEmbed
+ */
+class WPSEO_OpenGraph_OEmbed implements WPSEO_WordPress_Integration {
+	/**
+	 * Registers the hooks.
+	 */
+	public function register_hooks() {
+		add_filter( 'oembed_response_data', array( $this, 'wpseo_hook_oembed' ), 10, 4 );
+	}
+
+	/**
+	 * Callback function to pass to the oEmbed's response data that will enable
+	 * support for using the image and title set by the WordPress SEO plugin's fields. This
+	 * address the concern where some social channels/subscribed use oEmebed data over OpenGraph data
+	 * if both are present.
+	 *
+	 * @param array  $data   - The oEmbed data.
+	 * @param object $post   - The current Post object.
+	 * @param number $width  - The current post's width (more applicable for attachments).
+	 * @param number $height - The current post's height (more applicable for attachments).
+	 *
+	 * @see https://developer.wordpress.org/reference/hooks/oembed_response_data/ for hook info
+	 *
+	 * @return array $filter_data - An array of oEmbed data with modified values where appropriate.
+	 */
+	public function wpseo_hook_oembed( $data, $post, $width, $height ) {
+		// If OpenGraph is disabled, just return the data.
+		if ( ! WPSEO_Options::get( 'opengraph' ) ) {
+		   return $data;
+		}
+
+		// Data to be returned.
+		$filter_data = $data;
+
+		// Look for the Yoast meta values (image and title)...
+		$opengraph_title = WPSEO_Meta::get_value( 'opengraph-title', $post->ID );
+		$opengraph_image = WPSEO_Meta::get_value( 'opengraph-image', $post->ID );
+
+		// If yoast has a title set, update oEmbed with Yoast's title.
+		if ( ! empty( $yoast_title ) ) {
```

Comment:
I think that should be: `if ( ! empty( $opengraph_title ) ) {`,

Reply:
Yikes, slop! Yuck! Sorry about the lazy copy and paste there. I have fixed this.